src/actions/AppActions.ts

```diff
@@ -366,6 +366,16 @@ async function saveAll() {
 
 export async function deployAndRun(fiddleName: string, pageName: string = "", contractSuffix: string = "") {
   gaEvent(pageName === "test.html" ? "deployAndTest" : "deployAndRun");
+
+  // TODO: Remove ugly hack with window
```

Comment:
do we need to copy this todo everywhere? Let's just keep this, since I don't see a good way to solve this any time soon.

Reply:
it generally can be solved by moving state management to happen through `AppStore`, but yeah, easier to hack around for now